lib/client/client.go

```diff
@@ -577,58 +577,92 @@ func (client *NodeClient) ExecuteSCP(cmd scp.Command) error {
 	return trace.Wrap(err)
 }
 
-// listenAndForward listens on a given socket and forwards all incoming connections
-// to the given remote address via
-func (client *NodeClient) listenAndForward(socket net.Listener, remoteAddr string) {
-	defer socket.Close()
-	defer client.Close()
-	proxyConnection := func(incoming net.Conn) {
-		defer incoming.Close()
-		var (
-			conn net.Conn
-			err  error
-		)
-		log.Debugf("nodeClient.listenAndForward(%v -> %v) started", incoming.RemoteAddr(), remoteAddr)
-		for attempt := 1; attempt <= 5; attempt++ {
-			conn, err = client.Client.Dial("tcp", remoteAddr)
-			if err != nil {
-				log.Errorf("Connection attempt %v: %v", attempt, err)
-				// failed to establish an outbound connection? try again:
-				time.Sleep(time.Millisecond * time.Duration(100*attempt))
-				continue
-			}
-			// connection established: continue:
-			break
+func (client *NodeClient) proxyConnection(incoming net.Conn, remoteAddr string) {
+	defer incoming.Close()
+
+	var (
+		conn net.Conn
+		err  error
+	)
+	log.Debugf("nodeClient.proxyConnection(%v -> %v) started", incoming.RemoteAddr(), remoteAddr)
```

Comment:
use parameters here

Reply:
Changed the log line to the following:

```
log.Debugf("Attempting to connect proxy from %v to %v.", conn.RemoteAddr(), remoteAddr)
```

I think this reads much better. Using parameters would reduce readability significantly in this situation.